src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -93,6 +101,66 @@ export class ZerionBalancesApi implements IBalancesApi {
     }
   }
 
+  /**
+   * NOTE: Zerion does not support limit & offset parameters.
+   *
+   * It uses a "size" query param for the page size, and an "after" parameter for the offset.
+   * "size" is an integer which could be mapped to "limit", but "after" is a base64-encoded string.
+   *
+   * Since this setup does not align well with the CGW API, it is needed to encode/decode these parameters.
+   */
+  async getCollectibles(args: {
+    chainId: string;
+    safeAddress: string;
+    limit?: number;
+    offset?: number;
+  }): Promise<Page<Collectible>> {
+    try {
+      const cacheDir = CacheRouter.getZerionCollectiblesCacheDir(args);
+      const chainName = this._getChainName(args.chainId);
+      const url = `${this.baseUri}/v1/wallets/${args.safeAddress}/nft-positions`;
+      const pageAfter = this._encodeZerionPageOffset(args.offset);
+      const params = {
+        'filter[chain_ids]': chainName,
+        sort: ZerionBalancesApi.collectiblesSorting,
+        'page[size]': args.limit,
+        ...(pageAfter && { 'page[after]': pageAfter }),
+      };
+
+      const zerionCollectibles = await this.dataSource.get<ZerionCollectibles>({
+        cacheDir,
+        url,
+        notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
+        networkRequest: {
+          headers: { Authorization: `Basic ${this.apiKey}` },
+          params,
+        },
+        expireTimeSeconds: this.defaultExpirationTimeInSeconds,
+      });
+
+      // TODO: Zerion does not provide the items count. Change the Page entity to make count attribute nullable.
+      // Zerion does not provide a "previous" cursor.
+      return {
+        count: zerionCollectibles.data.length,
+        next: this._decodeZerionPagination(zerionCollectibles.links.next ?? ''),
+        previous: null,
+        results: this._mapCollectibles(zerionCollectibles.data),
+      };
+    } catch (error) {
+      throw new DataSourceError(
+        `Error getting ${args.safeAddress} collectibles from provider: ${asError(error).message}}`,
+      );
```

Comment:
Would it be better to use the `HttpErrorFactory` here instead?

Reply:
Yes! Good catch. Changed in [1a730ce](https://github.com/safe-global/safe-client-gateway/pull/1121/commits/1a730ce238b8101cd84728a16c784176c139b255)